openmc/deplete/batchwise.py

```diff
@@ -0,0 +1,1157 @@
+        Default to 'constant-volume'
+    bracketed_method : {'brentq', 'brenth', 'ridder', 'bisect'}, optional
+        Solution method to use.
+        This is equivalent to the `bracket_method` parameter of the
+        `search_for_keff`.
+        Defaults to 'brentq'.
+    tol : float
+        Tolerance for search_for_keff method.
+        This is equivalent to the `tol` parameter of the `search_for_keff`.
+        Default to 0.01
+    target : Real, optional
+        This is equivalent to the `target` parameter of the `search_for_keff`.
+        Default to 1.0.
+    print_iterations : Bool, Optional
+        Whether or not to print `search_for_keff` iterations.
+        Default to True
+    search_for_keff_output : Bool, Optional
+        Whether or not to print transport iterations during  `search_for_keff`.
+        Default to False
+    Attributes
+    ----------
+    cell : openmc.Cell or int or str
+        OpenMC Cell identifier to where apply batch wise scheme
+    attrib_name : str {'translation', 'rotation'}
+        Cell attribute type
+    axis : int {0,1,2}
+        Directional axis for geometrical parametrization, where 0, 1 and 2 stand
+        for 'x', 'y' and 'z', respectively.
+    samples : int
+        Number of samples used to generate volume estimates for stochastic
+        volume calculations.
+    """
+    def __init__(self, cell, attrib_name, operator, model, bracket,
+                 bracket_limit, axis, density_treatment='constant-volume',
+                 bracketed_method='brentq', tol=0.01, target=1.0, samples=1000000,
+                 print_iterations=True, search_for_keff_output=True):
+
+        super().__init__(cell, operator, model, bracket, bracket_limit, axis,
+                         density_treatment, bracketed_method, tol, target,
+                         print_iterations, search_for_keff_output)
+
+        check_value('attrib_name', attrib_name,
+                    ('rotation', 'translation'))
+        self.attrib_name = attrib_name
+
+        # check if cell is filled with 2 cells
+        check_length('fill materials', self.cell.fill.cells, 2)
+
+        # Initialize vector
+        self.vector = np.zeros(3)
+
+        self.cell_materials = [cell.fill for cell in \
+                        self.cell.fill.cells.values() if cell.fill.depletable]
+
+        check_type('samples', samples, int)
+        self.samples = samples
+
+        if self.cell_materials:
+            self._initialize_volume_calc()
+
+    def _get_cell_attrib(self):
+        """
+        Get cell attribute coefficient.
+        Returns
+        -------
+        coeff : float
+            cell coefficient
+        """
+        for cell in openmc.lib.cells.values():
+            if cell.id == self.cell.id:
+                if self.attrib_name == 'translation':
+                    return cell.translation[self.axis]
+                elif self.attrib_name == 'rotation':
+                    return cell.rotation[self.axis]
```

Comment:
For these calls when we're digging through the cells in the model, can we instead save the desired cell instance on this class? Maybe that isn't super ideal because we're something from the c++ side. But that's an awful lot of iterations over cells when a simple `self._cell` seems cheap and fast